### DISCUSSION TITLE
关于cgo部分代码ptid_comm的疑惑

### DISCUSSION THREAD

**Discussion by zcl1115** [harmonycloud/kindling#628] [agent collector]
如题所示，从代码里面观察到有个map ptid_comm用于存放pid的相关的信息，value是256char，不过好像没有观察到这个的清楚情况，就是这个map如果程序执行，是会一直扩张下去的吗？最终大小可能会是多少

**Comment by dxsup** [comment]
用于存放`线程ID`与`线程名`之间的对应关系，在Trace Profiling中会使用到，不启用Trace Profiling时不会存储数据。这个value代表线程名，占用字节数是变长的，最大是256byte，多数情况下线程名不会超过30个byte。

开启Trace Profiling功能时，在保存100k个线程数据的情况下，按照上限计算，256*100000/1024/1024=25MiB，算上map的额外内存占用，不超过30MiB。这里内存确实没有清理，是一个潜在的问题。

**Reply by zcl1115** [reply]
启动Trace Profiling与否的标志是怎么控制的呢

**Reply by dxsup** [reply]
需要手动通过一个前端页面来控制开启，默认都是关闭的。参见文档：https://kindlingx.com/docs/usage/enable-trace-profiling/

**Reply by zcl1115** [reply]
好的， 然后就是上个讨论中您有提到过jemalloc这个分析内存，想问下这个使用有什么文档说明吗？或者说大致流程是什么

**Reply by dxsup** [reply]
这里有一些[说明](https://github.com/jemalloc/jemalloc/wiki/Use-Case:-Leak-Checking), 主要还是得看jemalloc的文档了
